src/ucp/api/device/ucp_device_impl.h

```diff
@@ -115,6 +135,8 @@ ucp_device_prepare_multi(ucp_device_mem_list_handle_h mem_list_h,
  *
  * The routine returns a request that can be progressed and checked for
  * completion with @ref ucp_device_progress_req.
+ * It will only return when the message was posted or an error occurred.
+ * To do so, it might progress.
```

Comment:
`it` may be ambiguous since there are several nouns in the previous sentence.
`only` is in the wrong place.

`The routine returns only after the message has been posted or an error has occurred.`

Not sure about the second sentence. Since calling UCT progress is just a way of implementing the previous statement. it might be better to just delete the second sentence.